core/sdk.go

```diff
@@ -0,0 +1,197 @@
+package core
+
+import (
+	"context"
+
+	"github.com/dagger/dagger/dagql"
+)
+
+/*
+ClientGenerator is an interface that a module can implements to give
+client generation capabilities.
+
+The generated client is standalone and can be used in any project,
+even if no source code module is available.
+
+This interface MUST be implemented to support standalone client oriented
+features (`dagger client install`).
+*/
+type ClientGenerator interface {
+	/*
+		RequiredClientGenerationFiles returns the list of files that are required
```

Comment:
nit: should just be indented once?

Reply:
The formatter is auto formatting with 2 :/